src/evergreen/api.py

```diff
@@ -76,6 +76,7 @@ def __init__(
         auth: Optional[EvgAuth] = None,
         timeout: Optional[int] = None,
         session: Optional[requests.Session] = None,
+        debug: bool = False,
```

Comment:
If we do the below where we only guard the error level log statement, it might be best if we change this param to something more descriptive. Maybe something like `log_on_error`?